File: backend/src/domain/repositories/IUserRepository.ts
import { User } from '../entities/User';

export interface IUserRepository {
  findById(id: string): Promise<User | null>;
  findByEmail(email: string): Promise<User | null>;
  findByCommunity(communityId: string): Promise<User[]>;
  findVulnerableInArea(lat: number, lng: number, radiusKm: number): Promise<User[]>;
  findWithResourceInArea(resource: string, lat: number, lng: number, radiusKm: number): Promise<User[]>;
  save(user: User): Promise<User>;
  update(user: User): Promise<User>;
  delete(id: string): Promise<void>;
}

File: backend-dotnet/src/Hestia.Domain/Repositories/IUserRepository.cs
using Hestia.Domain.Entities;

namespace Hestia.Domain.Repositories;

public interface IUserRepository
{
    Task<User?> FindByIdAsync(Guid id, CancellationToken ct = default);
    Task<User?> FindByEmailAsync(string email, CancellationToken ct = default);
    Task<IReadOnlyList<User>> FindByCommunityAsync(Guid communityId, CancellationToken ct = default);
    Task<IReadOnlyList<User>> FindVulnerableInAreaAsync(double lat, double lng, double radiusKm, CancellationToken ct = default);
    Task<IReadOnlyList<User>> FindWithResourceInAreaAsync(string resource, double lat, double lng, double radiusKm, CancellationToken ct = default);
    Task<User> SaveAsync(User user, CancellationToken ct = default);
    Task<User> UpdateAsync(User user, CancellationToken ct = default);
    Task DeleteAsync(Guid id, CancellationToken ct = default);
}
